xbmc/input/InputManager.cpp

```diff
@@ -127,8 +127,15 @@ bool CInputManager::ProcessMouse(int windowId)
   if (g_application.WakeUpScreenSaverAndDPMS())
     return true;
 
-  // Retrieve the corresponding action
   CKey key(mousekey, (unsigned int)0);
+
+  // if this is a long press try to propagate ACTION_LONGCLICK to GUI
+  // before translating the button press (GUI controls might implement <onlongclick>)
+  if (key.GetButtonCode() == (KEY_MOUSE_LONG_CLICK + key.GetKeycode()) &&
+      g_application.OnAction(ACTION_LONGCLICK))
```

Comment:
I cannot say whether this change is okay as i have only very limited knowledge about our input system. But most probably @garbear can shed some light on this.

Reply:
Changed the implementation to work on any longclick on any mouse button. Unfortunately couldn't find any other way to do it. The button code is equal to `KEY_MOUSE_LONG_CLICK` + button number (which can have any value up to `MOUSE_MAX_BUTTON`. @garbear can you please take a look at this? Is the implementation ok?

Reply:
The early return here will likely cause problems. Can you set `mouseaction` to `ACTION_LONGCLICK` and process it as late as possible?

Reply:
I see what you mean, I've removed the early return both here and in touch longpress. As for the other suggestion - set `mouseaction` to `ACTION_LONGCLICK` and let it propagate I don't think it's a good idea. Here we want to let `ACTION_LONGCLICK` propagate to GUI before button translation happens. In case of mouse actions, long click are (unless changed by the user) mapped to "contextmenu". If you rewrite `mouseaction` `contextmenu` won't be triggered as you have effectively rewritten the action.
With the current implementation, Kodi propagates `ACTION_LONGCLICK` and then the buttontranslation for the window (in this case contextmenu) so I don't expect regressions here. The worst that may happen is that 2 actions are executed at the same time (one the result of longclick and the other the contextmenu) but that's really up to the skinner/addon implementation (and longclick is only implemented on a really small subset of controls that are not expected to ever have a context menu - they are not listitems).